#! /bin/sh /usr/share/dpatch/dpatch-run
## 40Fix_gtkiconload.dpatch by Jakub Wilk <jwilk@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Prevent program crashing when an icon is missing from the icon theme.

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' catfish-0.3.2~/catfish.py catfish-0.3.2/catfish.py
--- catfish-0.3.2~/catfish.py	2012-02-09 02:50:32.000000000 +0100
+++ catfish-0.3.2/catfish.py	2012-02-09 02:51:33.823351096 +0100
@@ -707,9 +707,12 @@
             return self.icon_cache[name]
         except KeyError:
             icon_size = gtk.icon_size_lookup(icon_size)[0]
-            icon = self.icon_theme.load_icon(name, icon_size, 0)
-            self.icon_cache[name] = icon
-            return icon
+            try:
+                icon = self.icon_theme.load_icon(name, icon_size, 0)
+                self.icon_cache[name] = icon
+                return icon
+            except gobject.GError:
+                return
 
     def get_thumbnail(self, path, icon_size=0, mime_type=None):
         """Try to fetch a small thumbnail."""
